network/p2p/p2pbuilder/libp2pNodeBuilder.go

```diff
@@ -551,73 +522,22 @@ func DefaultNodeBuilder(log zerolog.Logger,
-// - rsys: a routing.Routing object used to configure the GossipSub pubsub system.
-// - h: a libp2p host.Host object used to initialize the GossipSub pubsub system.
-//
-// Returns:
-// - p2p.PubSubAdapter: a GossipSub pubsub system for the libp2p node.
-// - error: if an error occurs during the creation of the GossipSub pubsub system, it is returned. Otherwise, nil is returned.
-// Note that on happy path, the returned error is nil. Any non-nil error indicates that the routing system could not be created
-// and is non-recoverable. In case of an error the node should be stopped.
-func (builder *LibP2PNodeBuilder) buildGossipSub(ctx context.Context, rsys routing.Routing, h host.Host) (p2p.PubSubAdapter, error) {
-	gossipSubConfigs := builder.gossipSubConfigFunc(&p2p.BasePubSubAdapterConfig{
-		MaxMessageSize: p2pnode.DefaultMaxPubSubMsgSize,
-	})
-	gossipSubConfigs.WithMessageIdFunction(utils.MessageID)
-	gossipSubConfigs.WithRoutingDiscovery(rsys)
-	if builder.subscriptionFilter != nil {
-		gossipSubConfigs.WithSubscriptionFilter(builder.subscriptionFilter)
-	}
-
-	var scoreOpt *scoring.ScoreOption
-	if builder.gossipSubPeerScoring {
-		scoreOpt = scoring.NewScoreOption(builder.logger, builder.idProvider, builder.peerScoringParameterOptions...)
-		gossipSubConfigs.WithScoreOption(scoreOpt)
-	}
-
-	gossipSubMetrics := p2pnode.NewGossipSubControlMessageMetrics(builder.metrics, builder.logger)
-	gossipSubConfigs.WithAppSpecificRpcInspector(func(from peer.ID, rpc *pubsub.RPC) error {
-		gossipSubMetrics.ObserveRPC(from, rpc)
-		return nil
-	})
-
-	if builder.gossipSubTracer != nil {
-		gossipSubConfigs.WithTracer(builder.gossipSubTracer)
-	}
-
-	gossipSub, err := builder.gossipSubFactory(ctx, builder.logger, h, gossipSubConfigs)
-	if err != nil {
-		return nil, fmt.Errorf("could not create gossipsub: %w", err)
-	}
-
-	if scoreOpt != nil {
-		scoreOpt.SetSubscriptionProvider(scoring.NewSubscriptionProvider(builder.logger, gossipSub))
-	}
-
-	return gossipSub, nil
-}
-
-// buildRouting creates a new routing system for a libp2p node using the provided host.
-// It returns the newly created routing system and any errors encountered during its creation.
+// buildRouting creates a new rsys system for a libp2p node using the provided host.
+// It returns the newly created rsys system and any errors encountered during its creation.
 //
 // Arguments:
 // - ctx: a context.Context object used to manage the lifecycle of the node.
-// - h: a libp2p host.Host object used to initialize the routing system.
+// - h: a libp2p host.Host object used to initialize the rsys system.
 //
 // Returns:
-// - routing.Routing: a routing system for the libp2p node.
-// - error: if an error occurs during the creation of the routing system, it is returned. Otherwise, nil is returned.
-// Note that on happy path, the returned error is nil. Any non-nil error indicates that the routing system could not be created
+// - rsys.Routing: a rsys system for the libp2p node.
```

Comment:
Great job catching that! The change you noticed was unintentional and has since been fixed in https://github.com/onflow/flow-go/pull/3986/commits/3ea3c075d63d580d3062df9f933907862db333d2.